### DISCUSSION TITLE
A and RHS

### DISCUSSION THREAD

**Discussion by SamL1u** [yalmip/YALMIP#1124] [General discussion] [answered]
Hello, John. I exported the model in yalmip through 'export', but I found that the number of rows of coefficient matrix A in the model was inconsistent with the number of constraints I wrote, and the number of columns of A was inconsistent with the number of variables I wrote. How can I get the A and RHS of the original model?

**Comment by johanlofberg** [comment, marked as answer]
That is impossible to answer without the model. If you  e.g. write abs(x)<=1 then you wrote 1 constraint, but the LP model will have 3 constraints (for x scalar).

**Reply by SamL1u** [reply]
Here are my codes.
[CCG_eng.txt](https://github.com/yalmip/YALMIP/files/8752640/CCG_eng.txt)
There are no nonlinear constraints in my constraints, so auxiliary variables should not be added.
There are 4157 columns(binvars+sdpvars) in model.A, but in fact, There are only 3833 variables defined in my model((binvars+sdpvars) as shown as follow,

Ness+Ness*Tf*2+Ndpot+(Ndpot+3)*Tf*Ncar+Ncar*Tf+Nload*Tf+Nbus*Tf+Nline*Tf*2+Nbus*Tf*2+Ness*Tf*4+Ness*(Tf+1)+Ness*2+Ndpot*Tf*Ncar*2+Ncar+Ndpot+Ness+Nfdst*Ndpot+3+Ncar+Ndpot

ans =

    3833

**Reply by johanlofberg** [reply]
Code does not run as you use local data

**Reply by SamL1u** [reply]
Here are the local data
[PtQt.zip](https://github.com/yalmip/YALMIP/files/8753264/PtQt.zip)

**Reply by johanlofberg** [reply]
the not operator is a high-level construct which introduces a bunch of auxilliary variables. You should just write 1-x in this trivial case

```
>> yalmip('clear')
>> x = binvar(1);
>> C = [~x == 0];
>> yalmip('nvars')

ans =

     3

```

not only will you save variables, your code will probably run a lot faster as it currently stalls on the generation of all those operators

**Reply by SamL1u** [reply]
Thank you, John.you solved the problem that bothered me for a few days.It's very nice of you.
